include/swift/SILOptimizer/Utils/PartitionUtils.h

```diff
@@ -351,6 +356,21 @@ class Partition {
     return p;
   }
 
+  static Partition separateRegions(ArrayRef<Element> indices) {
+    Partition p;
+    if (indices.empty())
+      return p;
+
+    Region max_index =
+        Region(*std::max_element(indices.begin(), indices.end()));
```

Comment:
Maybe `fresh_label` could be assigned after we went over indices, seems like `max_index` could be computed by that loop?

Reply:
Sure.